include/cache.h

```diff
@@ -13,48 +13,43 @@
 extern "C" {
 #endif
 
-#define _sys_cache_flush_sig(x) void (x)(vaddr_t virt, size_t size)
-
-#if defined(CONFIG_CACHE_FLUSHING)
-
-#if defined(CONFIG_ARCH_CACHE_FLUSH_DETECT)
-	typedef _sys_cache_flush_sig(_sys_cache_flush_t);
-	extern _sys_cache_flush_t *sys_cache_flush;
+static inline void sys_cache_flush(void *addr, size_t size, u32_t flags)
```

Comment:
These APIs are part of the Zephyr public interface and will need doxygen documentation.